fabtests/common/shared.c

```diff
@@ -474,6 +474,8 @@ int ft_reg_mr(struct fi_info *fi, void *buf, size_t size, uint64_t access,
 
 	ft_fill_mr_attr(&iov, &dmabuf, 1, access, key, iface, device, &attr, flags);
 	ret = fi_mr_regattr(domain, &attr, flags, mr);
+	if (opts.options & FT_OPT_REG_DMABUF_MR)
+		(void)close(dmabuf_fd);
```

Comment:
I think you need to introduce a general opt to handle the fd close, like https://github.com/ofiwg/libfabric/commit/5357ceecd708d7eedd72e2df292a7a6b55a9df58. It is possible that hmem iface has different API to make it.